app/src/main/java/ru/netology/custom_view/ui/StatsView.kt

```diff
@@ -110,26 +133,35 @@ class StatsView @JvmOverloads constructor(
     }
 
     override fun onDraw(canvas: Canvas) {
-        if (data.isEmpty()) {
+        if (data.isEmpty() || progress == 0F) {
             return
         }
-
-        var startFrom = -90F
+        var startFrom = INITIAL_ANGLE
+        val maxAngle = 360 * progress
         paint.color = colorOfEmptySpace
         canvas.drawCircle(center.x, center.y, radius, paint)
-
         for ((index, datum) in data.withIndex()) {
-
             val angle = if (sumValues > 100F) {
                 360F * (datum / sumValues)
             } else {
-                (datum * 3.6F)
+                datum * 3.6F
             }
-            paint.color = colors.getOrNull(index) ?: randomColor()
-            canvas.drawArc(oval, startFrom, angle, false, paint)
-            paint.color = colors[0]
-            canvas.drawArc(oval, -90F, 1F, false, paint)
 
+            if (startFrom - INITIAL_ANGLE + angle > maxAngle) {
+                drawData(
+                        index = index,
+                        canvas = canvas,
+                        startFrom = startFrom,
+                        sweepAngle = maxAngle - startFrom + INITIAL_ANGLE
+                )
+                return
+            }
+            drawData(
+                    index = index,
+                    canvas = canvas,
+                    startFrom = startFrom,
+                    sweepAngle = angle
```

Comment:
я бы заменил на универсальный вариант 
```suggestion
                    sweepAngle = min(angle, maxAngle - startFrom)
```
Тогда не понадобится отрисовка в `if`

Reply:
Опять же, если я уберу ветку if, то у меня не получится прервать onDraw

Reply:
А я не говорил убрать ветку `if` 😏 
Проверка на прерывание цикла никуда не девается - это абсолютно правильное решение.
Кстати, да - чтобы прервать цикл есть ключевое слово `break`. `break` прервёт цикл, но не вернёт из функции. Имейте это ввиду - может пригодиться

Reply:
исправлено